rllib/policy/dynamic_tf_policy.py

```diff
@@ -39,61 +46,87 @@ class DynamicTFPolicy(TFPolicy):
+                     [Policy, gym.spaces.Space, gym.spaces.Space,
+                      PolicyConfigDict], None]] = None,
+                 make_model: Optional[Callable[
+                     [Policy, gym.spaces.Space, gym.spaces.Space,
+                      PolicyConfigDict], ModelV2]] = None,
+                 action_sampler_fn: Callable[
+                     [TensorType, List[TensorType]], Tuple[
+                         TensorType, TensorType]] = None,
+                 action_distribution_fn: Optional[Callable[
+                     [Policy, ModelV2, TensorType, TensorType, TensorType],
+                     Tuple[TensorType, type, List[TensorType]]]] = None,
+                 existing_inputs: Optional[Dict[str, tf.placeholder]] = None,
+                 existing_model: Optional[ModelV2] = None,
+                 get_batch_divisibility_req: Optional[int] = None,
+                 obs_include_prev_action_reward: bool = True):
         """Initialize a dynamic TF policy.
 
         Arguments:
-            observation_space (gym.Space): Observation space of the policy.
-            action_space (gym.Space): Action space of the policy.
-            config (dict): Policy-specific configuration data.
-            loss_fn (func): function that returns a loss tensor the policy
-                graph, and dict of experience tensor placeholders
-            stats_fn (func): optional function that returns a dict of
-                TF fetches given the policy and batch input tensors
-            grad_stats_fn (func): optional function that returns a dict of
-                TF fetches given the policy and loss gradient tensors
-            before_loss_init (Optional[callable]): Optional function to run
-                prior to loss init that takes the same arguments as __init__.
-            make_model (func): optional function that returns a ModelV2 object
-                given (policy, obs_space, action_space, config).
+            observation_space (gym.spaces.Space): Observation space of the policy.
+            action_space (gym.spaces.Space): Action space of the policy.
+            config (PolicyConfigDict): Policy-specific configuration data.
+            loss_fn (Callable[[Policy, ModelV2, type, SampleBatch],
+                TensorType]): Function that returns a loss tensor the policy
+                graph, and dict of experience tensor placeholders.
+            stats_fn (Optional[Callable[[Policy, SampleBatch],
+                Dict[str, TensorType]]]): Optional function that returns a dict
+                of TF fetches given the policy and batch input tensors.
+            grad_stats_fn (Optional[Callable[[Policy, SampleBatch,
+                List[Tuple[TensorType, TensorType]]], Dict[str, TensorType]]]):
+                Optional function that returns a dict of TF fetches given the
+                policy and loss gradient tensors.
+            before_loss_init (Optional[Callable[
```

Comment:
Might be better to replace the datastructure of the input with an explanation of the input. (e.g. policy, action_space, obs_space), since the datastructure type is shown above in the arguments

Reply:
This function takes in other functions as input and is hence an exception compared to the labelling of the other methods in the later parts of your PR.

Reply:
+1